app/src/main/java/com/asfoundation/wallet/ui/iab/InAppPurchaseInteractor.java

```diff
@@ -261,12 +266,15 @@ private boolean shouldSwap(List<PaymentMethod> paymentMethods, int position) {
       } else if (id.equals("appcoins_credits") && !filteredGateways.contains(
           Gateway.Name.appcoins_credits)) {
         iterator.remove();
-      } else if (paymentMethod.getGateway()
-          .getName() == (Gateway.Name.myappcoins)
-          && paymentMethod.getAvailability() != null
-          && paymentMethod.getAvailability()
-          .equals("UNAVAILABLE")) {
-        iterator.remove();
+      } else {
+        if (paymentMethod.getGateway()
+            .getName() == (Gateway.Name.myappcoins)) {
+          paymentMethod.getAvailability();
+          if (paymentMethod.getAvailability()
+              .equals("UNAVAILABLE")) {
```

Comment:
Why did you change the logic here?

Reply:
Because paymentMethod.getAvailability() returns non null values always and thus the check is unnecessary